codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -233,35 +232,32 @@ void onUnknownFrame0(ChannelHandlerContext ctx, byte frameType, int streamId, Ht
         @Override
         public int onDataRead(final ChannelHandlerContext ctx, int streamId, ByteBuf data,
                 int padding, boolean endOfStream) throws Http2Exception {
-            // Check if we received a data frame for a stream which is half-closed
             Http2Stream stream = connection.requireStream(streamId);
```

Comment:
@Scottmitch @nmittler I was thinking about using `stream(streamId)` instead of `requireStream(streamId)` here and simply ignore the frame if the stream doesn't exist. My reasoning is that `requireStream(streamId)` raises a connection error and would thus close the whole connection. So e.g. if we receive a data frame for a stream that is already closed & garbage collected we would fail every stream on the connection. IMHO it's much more reasonable to just ignore this frame and continue processing the other streams. WDYT?